src/mesh/Foam2MooseMeshGen.C

```diff
@@ -239,11 +329,15 @@ Foam2MooseMeshAdapter::setUpSerial()
 void
 Foam2MooseMeshAdapter::setUpParallel()
 {
+  // _loc2glob should be a map from the local id to the global id and it is read directly from the
+  // pointProcAddressing file
   _loc2glob = getLocalGlobalMap(_mesh_wrapper.mesh());
   auto local_point = getLocalPoints<Foam::labelIOList>(*_loc2glob, _patch_id, _mesh_wrapper);
   std::vector<int32_t> face_count, face_point_id, face_subdomain_id;
   auto face_info = getLocalFaceInfo<Foam::labelIOList>(*_loc2glob, _mesh_wrapper, _patch_id);
   gatherUniquePoints(local_point);
+  getMissingPoints(_point, *_loc2glob, _patch_id, _mesh_wrapper, _comm);
```

Comment:
Might be good to have a comment explaining why we might have "missing" points in the first place, and that this is a workaround that is necessary to make things work.